src/Abstractions/TaskName.cs

```diff
@@ -148,7 +149,13 @@ public override bool Equals(object? obj)
   /// <returns>A 32-bit hash code value.</returns>
   public override int GetHashCode()
   {
-    return StringComparer.OrdinalIgnoreCase.GetHashCode(this.Name);
+    if (string.IsNullOrEmpty(this.Version))
+    {
+      return StringComparer.OrdinalIgnoreCase.GetHashCode(this.Name);
+    }
+
+    return StringComparer.OrdinalIgnoreCase.GetHashCode(this.Name) * 31
```

Comment:
I think you can add a reference to [Microsoft.Bcl.HashCode](https://www.nuget.org/packages/Microsoft.Bcl.HashCode). You can then use `HashCode.Combine(this.Name, this.Version);`

Reply:
I referenced the code in src\Shared\Core\CSharpLangCompat.cs 

```c#
        public override int GetHashCode()
        {
            return (this.Start.GetHashCode() * 31) + this.End.GetHashCode();
        }
```

I agree with you that [Microsoft.Bcl.HashCode](https://www.nuget.org/packages/Microsoft.Bcl.HashCode) is a good  choose, but it means that we need to introduce a new dependency. 

If you and Chris agree that introducing new dependency in this PR is acceptable, I would like to update the code.

Reply:
Done. Added dependency and update code as your suggestion.